### PR TITLE
SALTO-4321: Include/ Exclude list doesn't work for private API types

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Fix a bug causing config validation error when trying to include / exclude a type defined with ducktype api definitions (=Okta private API types)

---
_Release Notes_: 

_Okta adapter_:
-  Fix a bug in `include` and `exclude` for Okta types

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/src/adapter_creator.ts [hunk 1]
```diff
@@ -43,6 +45,22 @@ const credentialsFromConfig = (config: Readonly<InstanceElement>): Credentials =
...
 
+const validateOktaFetchConfig = (
+  fetchConfig: OktaFetchConfig,
+  clientConfig: OktaClientConfig,
+  apiDefinitions: OktaSwaggerApiConfig,
+  privateApiDefinitions: OktaDuckTypeApiConfig,
+): void => {
+  const supportedTypes = clientConfig.usePrivateAPI === false
+    ? Object.keys(apiDefinitions.supportedTypes)
+    : Object.keys(apiDefinitions.supportedTypes).concat(Object.values(privateApiDefinitions.supportedTypes).flat())
```
Comment (netama) [inline packages/okta-adapter/src/adapter_creator.ts:56]: why is one `keys` and one `values`?

File: packages/okta-adapter/src/adapter_creator.ts [hunk 2]
```diff
@@ -43,6 +45,22 @@ const credentialsFromConfig = (config: Readonly<InstanceElement>): Credentials =
   }
 }
 
+const validateOktaFetchConfig = (
+  fetchConfig: OktaFetchConfig,
+  clientConfig: OktaClientConfig,
+  apiDefinitions: OktaSwaggerApiConfig,
+  privateApiDefinitions: OktaDuckTypeApiConfig,
+): void => {
```
Comment (netama) [inline packages/okta-adapter/src/adapter_creator.ts:53]: nit: maybe use named parameters to avoid mistakes? (since this is not exported it's not critical)

**Review by netama — APPROVED** [review]
LGTM, added a question

File: packages/okta-adapter/src/config.ts [hunk 3]
```diff
@@ -1523,14 +1523,14 @@ const DUCKTYPE_TYPES: OktaDuckTypeApiConfig['types'] = {
...
-  DisplayLanguageSettings: ['DisplayLanguage'],
-  ReauthenticationSettings: ['Reauthentication'],
+  EmailNotifications: ['EmailNotifications'],
+  EndUserSupport: ['EndUserSupport'],
+  ThirdPartyAdmin: ['ThirdPartyAdmin'],
+  EmbeddedSignInSuppport: ['EmbeddedSignInSuppport'],
+  SignOutPage: ['SignOutPage'],
+  BrowserPlugin: ['BrowserPlugin'],
+  DisplayLanguage: ['DisplayLanguage'],
+  Reauthentication: ['Reauthentication'],
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/config.ts:1533]: `Include` didn't work with the types as it was, so changing this solved it

File: packages/okta-adapter/src/adapter_creator.ts [hunk 4]
```diff
@@ -43,6 +45,27 @@ const credentialsFromConfig = (config: Readonly<InstanceElement>): Credentials =
   }
 }
 
+const validateOktaFetchConfig = ({
+  fetchConfig,
+  clientConfig,
+  apiDefinitions,
+  privateApiDefinitions,
+}:
+{ fetchConfig: OktaFetchConfig
```
Comment (netama) [inline packages/okta-adapter/src/adapter_creator.ts:54]: indentation nit (start and finish should be aligned):
```suggestion
}: { fetchConfig: OktaFetchConfig
```

**Review by netama — APPROVED** [review]
🎈

**Pull request merged**